### ISSUE TITLE
asyncio.TimeoutError does not release the request

### ISSUE THREAD

**Issue by Insoleet** [aio-libs/aiohttp#612]
Test showing the bug, based on the API available in test_web_functional : 

``` python
    def test_simple_timeout(self):

        @asyncio.coroutine
        def handler(request):
            body = yield from request.read()
            self.assertEqual(b'', body)
            yield from asyncio.sleep(2, loop=self.loop)
            return web.Response(body=b'OK')

        @asyncio.coroutine
        def go():
            _, srv, url = yield from self.create_server('GET', '/', handler)
            try:
                resp = yield from asyncio.wait_for(request('GET', url, loop=self.loop), 1, loop=self.loop)
            except TimeoutError:
                pass
            yield from asyncio.sleep(3, loop=self.loop)
```

Results in the following error : 

```
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x7efe1383c048>
Task was destroyed but it is pending!
task: <Task pending coro=<__iter__() running at /home/inso/code/aiohttp/aiohttp/client.py:475> wait_for=<Future pending cb=[BaseSelectorEventLoop._sock_connect_done(8)(), Task._wakeup()]>>
```

**Comment by asvetlov** [comment]
The error is not about unreleased request but not closed session.
Should be fixed anyway.

**Comment by Insoleet** [comment]
Out of curiosity, what is the difference ? Just so that I better understand how aiohttp works !

Was it fixed some time ago ? I can't find any commit about this bug.

**Comment by asvetlov** [comment]
It's not fixed yet but we should add a fix.
session can be used for handling multiple requests, keep-alives and cookies: http://aiohttp.readthedocs.org/en/stable/client.html#keep-alive-connection-pooling-and-cookie-sharing
It also support the limit of opened connections via `Connector`: http://aiohttp.readthedocs.org/en/stable/client.html#limiting-connection-pool-size

`Response` is just HTTP response for HTTP request.

Calls like `request`, `get` etc. create temporary `ClientSession` internally.

**Comment by asvetlov** [comment]
Fixed by #611

**Comment by asvetlov** [comment]
Fixed by #707
